src/Symfony/Component/Config/Resource/DirectoryResource.php

```diff
@@ -87,7 +87,12 @@ public function isFresh($timestamp)
                 continue;
             }
 
-            $newestMTime = max($file->getMTime(), $newestMTime);
+            // early return if any file mtime is greater than passed timestamp
+            if (($fileMTime = $file->getMTime()) > $timestamp) {
+                return false;
+            }
+
+            $newestMTime = max($fileMTime, $newestMTime);
```

Comment:
Let me know if I'm wrong, but I think the test is failing in this case because the implementation and the test case are slightly flawed.

Try:

```diff
diff --git a/src/Symfony/Component/Config/Resource/DirectoryResource.php b/src/Symfony/Component/Config/Resource/DirectoryResource.php
index dcb5e19..bef0b2b 100644
--- a/src/Symfony/Component/Config/Resource/DirectoryResource.php
+++ b/src/Symfony/Component/Config/Resource/DirectoryResource.php
@@ -95,7 +95,7 @@ class DirectoryResource implements SelfCheckingResourceInterface, \Serializable
             $newestMTime = max($fileMTime, $newestMTime);
         }
 
-        return $newestMTime < $timestamp;
+        return $newestMTime <= $timestamp;
     }
 
     public function serialize()
```

and the test case should fail. However, I think a resource should be considered fresh if its mtime is the exact same as provided timestamp (on the contrary, you should revised the new conditions to use `$fileMTime >= $timestamp`).

The following for instance should work:

<details>
<summary>See patch</summary>

```diff
diff --git a/src/Symfony/Component/Config/Resource/DirectoryResource.php b/src/Symfony/Component/Config/Resource/DirectoryResource.php
index dcb5e19..deb40b0 100644
--- a/src/Symfony/Component/Config/Resource/DirectoryResource.php
+++ b/src/Symfony/Component/Config/Resource/DirectoryResource.php
@@ -74,7 +74,10 @@ class DirectoryResource implements SelfCheckingResourceInterface, \Serializable
             return false;
         }
 
-        $newestMTime = filemtime($this->resource);
+        if($timestamp < $newestMTime = filemtime($this->resource)) {
+            return false;
+        };
+
         foreach (new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->resource), \RecursiveIteratorIterator::SELF_FIRST) as $file) {
             // if regex filtering is enabled only check matching files
             if ($this->pattern && $file->isFile() && !preg_match($this->pattern, $file->getBasename())) {
@@ -88,14 +91,12 @@ class DirectoryResource implements SelfCheckingResourceInterface, \Serializable
             }
 
             // early return if any file mtime is greater than passed timestamp
-            if (($fileMTime = $file->getMTime()) > $timestamp) {
+            if ($timestamp < $fileMTime = $file->getMTime()) {
                 return false;
             }
-
-            $newestMTime = max($fileMTime, $newestMTime);
         }
 
-        return $newestMTime < $timestamp;
+        return true;
     }
 
     public function serialize()
diff --git a/src/Symfony/Component/Config/Tests/Resource/DirectoryResourceTest.php b/src/Symfony/Component/Config/Tests/Resource/DirectoryResourceTest.php
index a22209d..9d8e055 100644
--- a/src/Symfony/Component/Config/Tests/Resource/DirectoryResourceTest.php
+++ b/src/Symfony/Component/Config/Tests/Resource/DirectoryResourceTest.php
@@ -110,8 +110,10 @@ class DirectoryResourceTest extends \PHPUnit_Framework_TestCase
     public function testIsFreshDeleteFile()
     {
         $resource = new DirectoryResource($this->directory);
+        $time = time();
+        sleep(1);
         unlink($this->directory.'/tmp.xml');
-        $this->assertFalse($resource->isFresh(time()), '->isFresh() returns false if an existing file is removed');
+        $this->assertFalse($resource->isFresh($time), '->isFresh() returns false if an existing file is removed');
     }
 
     public function testIsFreshDeleteDirectory()
```

</details>